### PR TITLE
Gurobi inclusion

### PR DESCRIPTION
**Pull request by liangend**

### CODE REVIEW

File: LPChoose.jl [hunk 1]
```diff
@@ -1,12 +1,17 @@
 #include("LPChoose.jl"); LPChoose("smalldata.txt",100,0.00);
 using DataFrames,LinearAlgebra,CSV,DelimitedFiles,SparseArrays,GLPK,JuMP,Statistics, StatsBase
 using ProgressMeter
+
+import_Gurobi = false  # import Gurobi package when import_Gurobi = true
+if import_Gurobi == true
+    using Gurobi
```
Comment (reworkhow) [inline LPChoose.jl]:
```
try
    @eval import Gurobi
    println("Gurobi has been installed.")
catch
    println("Gurobi is not installed. GLPK has been installed.")
end
```

File: LPChoose.jl [hunk 2]
```diff
@@ -31,14 +36,15 @@ using ProgressMeter
       number of unique haplotypes in the population; "haplotype frequency" tends to cover the largest proportion of genomes in the population.
     * If `sequencing_homozygous_haplotypes_only`=`true`, LPChoose will only focus on sequencing homozygous haplotype
       segments to achieve a reduction in cost with an added benefit of phasing variant calls efficiently (Bickhart et al. 2015).
-
+    * ind_hap_lim is the maximum number of indentical haplotypes shared by selected individuals in each step for application 2, default value equal to 2.
```
Comment (reworkhow) [inline LPChoose.jl]: identical

File: LPChoose.jl [hunk 3]
```diff
@@ -60,7 +66,12 @@ function LPChoose(hapblock,budget="unlimited",MAF=0.0;
         println("-------containing all unique haplotypes----------")
         println("-----------RUN LINEAR PROGRAMMING-----------------\n")
         importance = ones(nind)
-        model = Model(GLPK.Optimizer)
+        if use_Gurobi == false
+            model = Model(GLPK.Optimizer)
+            set_optimizer_attribute(model, "tm_lim", 30 * 60 * 1_000)  # time limit in milliseconds (default 30 min)
+        else
```
Comment (reworkhow) [inline LPChoose.jl:78]: make "30" an argument for the function as "time_limit".

**Review by reworkhow — CHANGES_REQUESTED** [review]

File: LPChoose.jl (hunk 1 above)
Comment (reworkhow) [reply LPChoose.jl]: consider a better printout here to give users some suggestion which solver to use.

**Comment by liangend** [comment]
Updated Gurobi inclusion

**Review by liangend — COMMENTED** [review]
Updated Gurobi inclusion

**Review by liangend — COMMENTED** [review]
print out information

File: LPChoose.jl [hunk 4]
```diff
@@ -1,12 +1,22 @@
 #include("LPChoose.jl"); LPChoose("smalldata.txt",100,0.00);
 using DataFrames,LinearAlgebra,CSV,DelimitedFiles,SparseArrays,GLPK,JuMP,Statistics, StatsBase
 using ProgressMeter
+
+import_Gurobi = false  # import Gurobi package when import_Gurobi = true
```
Comment (reworkhow) [inline LPChoose.jl:5]: We don't need import_Gurobi. I don't see you provide an interface for users for import_Gurobi. If it is available, we just load it.

**Review by reworkhow — APPROVED** [review]

**Pull request merged**